packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External Dependencies
+ */
+
+import { uniqBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	withNotices,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	mediaUpload,
+	MediaUpload,
+} from '@wordpress/editor';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.onUploadFiles = this.onUploadFiles.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			isEditing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	uploadFromFiles( event ) {
+		this.onUploadFiles( event.target.files );
```

Comment:
Curious: Is there ever a chance this can be called with an empty `event.target.files` ? If so, how would we handle that?

Reply:
I can wrap this in a condition if it exists. I cant think of an instance where this would happen as upload from files is triggered after files are selected. Worth the condition I think

Reply:
Being defensive is good, but having an understanding of the types of values we're expecting to handle is better.